server/influx.go

```diff
@@ -173,26 +173,50 @@ func (s *Service) Write(w http.ResponseWriter, r *http.Request) {
 
 // setupQueryFromCommand set query parameters from its command
 func setupQueryFromCommand(req *chronograf.Query) {
-	// allow to set active database with USE command, examples:
+	// normalize whitespaces
+	req.Command = strings.Join(strings.Fields(req.Command), " ")
+
+	// set active database (and retention policy) from the query
+	useDb := func(dbSpec string) error {
+		dbSpecReader := csv.NewReader(bytes.NewReader(([]byte)(dbSpec)))
+		dbSpecReader.Comma = '.'
+		if dbrp, err := dbSpecReader.Read(); err == nil {
+			if len(dbrp) > 0 {
+				req.DB = dbrp[0]
+			}
+			if len(dbrp) > 1 {
+				req.RP = dbrp[1]
+			}
+			return nil
+		} else {
+			return err
+		}
+	}
+
+	// allow to set active database with USE command or via ON clause, examples:
 	//  use mydb
 	//  use "mydb"
 	//  USE "mydb"."myrp"
 	//  use "mydb.myrp"
 	//  use mydb.myrp
+	//  show tag keys on "mydb"
+	//  SHOW TAG KEYS ON "mydb"
 	if strings.HasPrefix(req.Command, "use ") || strings.HasPrefix(req.Command, "USE ") {
```

Comment:
To be safe with lower and upper cases, we can use something like:

```suggestion
	//  SHOW TAG KEYS ON "mydb"
	command := strings.ToLower(req.Command)
	if strings.HasPrefix(command, "use ") {
```